src/utilities/network-information-utils.test.ts

```diff
@@ -0,0 +1,123 @@
+import { NavigatorConnectionVariants } from "../enumerations/navigator-connection-variants";
+import { NetworkInformationUtils } from "./network-information-utils";
+
+describe("NetworkInformationUtils", () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Mocks
+    // -----------------------------------------------------------------------------------------
+
+    const mockWindowNavigator = (options?: {
```

Comment:
Is there a typed interface we get for `window.navigator` that has these? Would prefer that over defining the structure inline. If not, no worries